Blish HUD/_Utils/ClipboardUtil.cs

```diff
@@ -19,6 +26,43 @@ public static class ClipboardUtil {
         /// used to make async calls to the Windows clipboard.
         /// </summary>
         public static WindowsClipboardService WindowsClipboardService => _clipboardService;
-
+        /// <summary>
+        /// Returns the clipboard's content as a Dictionary of objects indexed by their data format or null if the clipboard is empty.
+        /// </summary>
+        /// <returns>Dictionary of objects indexed by their DataFormat (System.Windows.Forms.DataFormats).</returns>
+        public static Dictionary<string, object> GetClipboardContent() {
+            try {
+                var dataObject = Clipboard.GetDataObject();
+                if (dataObject == null) return null;
+                var formats = dataObject.GetFormats(false);
+                var backup = new Dictionary<string, object>();
+                foreach (var format in formats) 
+                    backup.Add(format, dataObject.GetData(format, false));
+                return backup;
+            } catch (ExternalException e) {
+                Logger.Error("Data could not be retrieved from the Clipboard. This typically occurs when the Clipboard is being used by another process. " + e.Message);
+            } catch (ThreadStateException e) {
+                Logger.Error("The current thread is not in single-threaded apartment (STA) mode and the MessageLoop property value is true. Add the STAThreadAttribute to your application's Main method. " + e.Message);
+            }
+            return null;
+        }
+        /// <summary>
+        /// Sets the clipboard's content by assigning it a new DataObject with the given content.
+        /// </summary>
+        /// <param name="content">Dictionary of objects indexed by their DataFormat (System.Windows.Forms.DataFormats).</param>
+        public static void SetClipboardContent(Dictionary<string, object> content) {
```

Comment:
The async clipboard library we're using for this should actually be handling this for us so that we don't have to interface with the clipboard directly ourselves.  https://github.com/undergroundwires/AsyncWindowsClipboard

Reply:
these were test functions to backup the clipboard. The asyncwindowclipboard doesn't support all DataFormats and I am assuming this is because you shouldn't try to restore a clipboard except string type (which asyncclipboard sole support lies.)
Will be removing these functions again.